crates/algokit_utils/src/transactions/transaction_ext.rs

```diff
@@ -0,0 +1,212 @@
+use algokit_transact::{
+    Address, ApplicationCallTransactionFields, AssetConfigTransactionFields,
+    AssetFreezeTransactionFields, AssetTransferTransactionFields, KeyRegistrationTransactionFields,
+    PaymentTransactionFields, Transaction,
+};
+
+/// Extension trait providing accessor methods for Transaction enum variants
+pub trait TransactionExt {
```

Comment:
This is an equivalent to what py does with the transaction wrapper abstraction and what ts does with an interface and a bunch of optionals to give ability to quickly 'parse' to a concrete typed transaction from response. If you think its better to also move this into transact crate i could tweak it further